src/widget/wcuemenupopup.cpp

```diff
@@ -0,0 +1,130 @@
+#include "widget/wcuemenupopup.h"
+
+#include <QHBoxLayout>
+#include <QInputDialog>
+#include <QVBoxLayout>
+
+#include "engine/engine.h"
+#include "util/color/color.h"
+
+WCueMenuPopup::WCueMenuPopup(QWidget* parent)
+        : QWidget(parent) {
+    QWidget::hide();
+    setWindowFlags(Qt::Popup);
+    setObjectName("WCueMenuPopup");
+
+    m_pCueNumber = new QLabel(this);
+    m_pCueNumber->setToolTip(tr("cue number"));
+    m_pCueNumber->setObjectName("CueNumberLabel");
+    m_pCueNumber->setAlignment(Qt::AlignLeft);
+
+    m_pCuePosition = new QLabel(this);
+    m_pCuePosition->setToolTip(tr("Cue position"));
+    m_pCuePosition->setObjectName("CuePositionLabel");
+    m_pCuePosition->setAlignment(Qt::AlignRight);
+
+    m_pEditLabel = new QLineEdit(this);
+    m_pEditLabel->setToolTip(tr("Edit cue label"));
+    m_pEditLabel->setObjectName("CueLabelEdit");
+    connect(m_pEditLabel, &QLineEdit::textEdited, this, &WCueMenuPopup::slotEditLabel);
+    connect(m_pEditLabel, &QLineEdit::returnPressed, this, &WCueMenuPopup::hide);
+
+    m_pColorPicker = new WColorPicker(this);
+    m_pColorPicker->setObjectName("CueColorPicker");
+    connect(m_pColorPicker, &WColorPicker::colorPicked, this, &WCueMenuPopup::slotChangeCueColor);
+
+    m_pRemoveCue = new QPushButton("", this);
+    m_pRemoveCue->setToolTip(tr("Remove this cue point"));
+    m_pRemoveCue->setFixedHeight(m_pEditLabel->sizeHint().height());
+    m_pRemoveCue->setObjectName("CueRemoveButton");
```

Comment:
I'm not sure. We could introduce a `default_style.qss` that will be applied before any skin's `style.css` (so that they can override it), but I'd rather do that in another PR.

Reply:
Implemented in https://github.com/mixxxdj/mixxx/pull/2362/commits/5649646eda50076839a80682645d88dad406edfa